### PR TITLE
[SYCL-MLIR][NFC] Adapt `IntegerRangeAnalysis` changes for upstreaming

### PR DESCRIPTION
**Pull request by victor-eds**
- Remove not needed headers being included
- Fix `check-mlir` due to variable being used in label in test

### CODE REVIEW

File: mlir/test/Dialect/Arith/int-range-interface.mlir [hunk 1]
```diff
@@ -731,18 +731,23 @@ func.func @extui_uses_unsigned(%arg0 : i32) -> i1 {
...
-/// Catch a bug that crash in getLoopBoundFromFold when
-/// SparseConstantPropagation loaded in the solver.
-
-// CHECK-LABEL: func.func @caller(%[[ARG0:.*]]
-// CHECK: call @callee(%[[ARG0]])
-// CHECK-LABEL: func.func @callee(%[[ARG0]]
-// CHECK: %[[LOAD:.*]] = affine.load %[[ARG0]]
-// CHECK: scf.for {{.*}} to %[[LOAD]] 
+/// Catch a bug that caused a crash in getLoopBoundFromFold when
+// SparseConstantPropagation is loaded in the solver.
```
Comment (jopperm) [inline mlir/test/Dialect/Arith/int-range-interface.mlir:735]: UltraNIT :tm: : why remove the third slash?

File: mlir/test/Dialect/Arith/int-range-interface.mlir [hunk 2]
```diff
@@ -731,18 +731,23 @@ func.func @extui_uses_unsigned(%arg0 : i32) -> i1 {
...
 }
 
-/// Catch a bug that crash in getLoopBoundFromFold when
-/// SparseConstantPropagation loaded in the solver.
-
-// CHECK-LABEL: func.func @caller(%[[ARG0:.*]]
-// CHECK: call @callee(%[[ARG0]])
-// CHECK-LABEL: func.func @callee(%[[ARG0]]
-// CHECK: %[[LOAD:.*]] = affine.load %[[ARG0]]
-// CHECK: scf.for {{.*}} to %[[LOAD]]
```
Comment (jopperm) [inline mlir/test/Dialect/Arith/int-range-interface.mlir:741]: This is just cleaning up parts of the test that were removed earlier, right?

**Review by jopperm — APPROVED** [review]

File: mlir/test/Dialect/Arith/int-range-interface.mlir (hunk 1 above)
Comment (victor-eds) [reply mlir/test/Dialect/Arith/int-range-interface.mlir:735]: #unintentional

File: mlir/test/Dialect/Arith/int-range-interface.mlir (hunk 2 above)
Comment (victor-eds) [reply mlir/test/Dialect/Arith/int-range-interface.mlir:741]: Yep, adapting to what the automatic tool generates. Previous was wrong as you cannot have variables in labels in FileCheck

**Pull request merged**